src/lib/statistics/statistics_objects/equal_distinct_count_histogram.cpp

```diff
@@ -49,11 +50,36 @@ template <typename T>
 std::vector<std::pair<T, HistogramCountType>> value_distribution_from_column(const Table& table,
                                                                              const ColumnID column_id,
                                                                              const HistogramDomain<T>& domain) {
-  // TODO(anybody) If you want to look into performance, this would probably benefit greatly from monotonic buffer
-  //               resources.
-  ValueDistributionMap<T> value_distribution_map;
-
+  // Since this function can be the main bottleneck for loading data in the benchmarks, we put some effort in
+  // optimizing it. For once, we try to get the distinct count in case we have dictionary-encoded data in place. This
+  // allows us to pre-size the robin_map which helps a lot with performance (no resizes for large maps) and memory
+  // consumption (as we can avoid pre-sizes all maps with the table size, which is often by far too large).
```

Comment:
_pre-allocating all maps_? Also, I don't know if we need this second memory consumption reason because it targets another optimization, allocating the maps with the table size, which is not the default case of using the map without any pre-allocation.